### PR TITLE
Add contributing.md with GH workflow and doc edits/creation details

### PR DESCRIPTION
**Pull request by LZoog**
If we end up moving this repository back into `experimenter` we can create a follow up ticket to make any needed doc updates in this section then.

Closes mozilla/experimenter#4722 / EXP-1027

### CODE REVIEW

File: docs/contributing.md
Comment (LZoog) [inline docs/contributing.md:24]: @jodyheavener I could've added a class or two here instead of `style={{` but I'm not sure that it'd be worth enforcing "no inline styles!!" in the doc hub. It'd be one more thing for reviewers to have to know to lookout for. Whatchu think?

File: docs/contributing.md
Comment (LZoog) [inline docs/contributing.md:97]: Oooh, @jodyheavener pointed out to me these aren't mobile-friendly, d'oh... I'll fix this up later today.

**Comment by LZoog** [comment]
Oh also! I want to modify the front-page README regarding this, and I'll go ahead and add a section for Experimenter engineers to set @jaredlockhart up to add docs related to the new publish status as discussed yesterday.

File: src/css/custom.css
Comment (jodyheavener) [inline src/css/custom.css:60]: I wonder how frequently we're going to see CSS changes to accommodate doc layouts 🤔 Is it worth adding a little blurb about how and where to add CSS?

File: docs/contributing.md
Comment (jodyheavener) [inline docs/contributing.md:59]: More of a nitpick than anything, but these aren't really quotes. I gather you're using `>` to make it stand out, but maybe we could use an admonition or some other button-y type element?

File: docs/contributing.md
Comment (jodyheavener) [inline docs/contributing.md:15]: Let's try and use as little HTML as possible in Markdown files (this can just be `**Notifications**`)

File: docs/contributing.md
Comment (jodyheavener) [inline docs/contributing.md:55]: I'm not sure it's not in the GFM docs, but Docusaurus supports proper Markdown [footnote syntax](https://www.markdownguide.org/extended-syntax#footnotes) that allows you to link directly to your footnote (and for the footnote to link back to the content). Can we use that instead?

File: docs/contributing.md
Comment (jodyheavener) [inline docs/contributing.md:166]: Nitpick: the `<hr>` in Dark mode could be a little more subtle if you felt like toning it down in the CSS :)

File: docs/contributing.md
Comment (jodyheavener) [inline docs/contributing.md:171]: Hrrmm, I kinda have beef with these custom spacing styles. Ideally the docs we write about how to write docs can be easily replicated, and using custom margins on images could be a barrier for someone who wants to make their contribution look like ours, and it also doesn't play well with mobile. Could these styles be added to the CSS as a class like "extra-padding" or "align-right"?

File: docs/contributing.md
Comment (jodyheavener) [reply docs/contributing.md:59]: It would also be good to mention/link to [admonitions](https://v2.docusaurus.io/docs/markdown-features/admonitions) somewhere, maybe near the bit about GFM

File: docs/contributing.md
Comment (jodyheavener) [inline docs/contributing.md:184]: Optional, but could be useful here and good to know for future code snippets: Docusaurus supports snippet titles: https://v2.docusaurus.io/docs/markdown-features/code-blocks#code-title

(I'll let you decide if this is something we should document, I'm not sure how widely used it'd be)

**Review by jodyheavener — COMMENTED** [review]
I need to go back through the content, but here are some initial thoughts. This is look really great BTW. I love how you've arranged the images.

File: docs/contributing.md
Comment (LZoog) [reply docs/contributing.md:59]: I tried using an admonition here but it honestly seemed to stand out _too_ much. 😆 I get what you mean about it not being a quote though so maybe I should use one just to be proper, like `tip` where I can add a custom title. Good call out on linking that doc.

File: docs/contributing.md
Comment (k88hudson) [inline docs/contributing.md:35]: Maybe, `If you aren't as familiar with git, we recommend using the GitHub user interface to make changes to docs.`

File: docs/contributing.md
Comment (k88hudson) [inline docs/contributing.md:79]: ❤️ this quick links

**Comment by k88hudson** [comment]
Yeah so this is super cool, love the links to "do this in github" especially. Definitely worth checking if it makes sense to some folks who will be using the github UI.

I think this could use some guidance around like what the process for asking for a review is / how things will get merged

File: docs/contributing.md
Comment (LZoog) [reply docs/contributing.md:15]: Yep agreed, missed this

File: docs/contributing.md
Comment (lmorchard) [reply docs/contributing.md:24]: Random notion off the top of my head: Rather than enforce no inline styles, I'd ask if this could have been a class that would come in handy to re-use later. Can also be an optimization to perform later, if a common style comes up often

Comment (LZoog) [reply docs/contributing.md:24]: Yeah, it's possible. I'm not sure how often folks will actually be using JSX in MD but I'll go ahead and convert these to classes.

File: docs/contributing.md
Comment (LZoog) [reply docs/contributing.md:29]: I'll rephrase. I mean pulling down the repo locally.

File: docs/contributing.md
Comment (LZoog) [reply docs/contributing.md:184]: _This is cool_

**Comment by LZoog** [comment]
Thank you for the comments everyone! I've pushed some updates and believe I've addressed all comments except:

> I think this could use some guidance around like what the process for asking for a review is / how things will get merged

@k88hudson we talked briefly in Slack about this - how do you feel about asking users to come back to their PRs to merge them after someone r+'s? We oftentimes approve things with a comment/suggestion or two that we consider non-blocking, so that seems more ideal in case they want to make a change before merging, but if someone isn't used to hanging out in GitHub, PRs could potentially hang out longer than they need to. As @jodyheavener and I said in Slack, perhaps it's OK to require this at first and see how things go?

Also, should users request the `project-nimbus` team under "Reviewers" (in which case, how would a user request to be added to this team)? Or, do we expect that users that want to receive notifications have configured them for this repo?

File: docs/contributing.md
Comment (LZoog) [reply docs/contributing.md]: OK after converting all these to classes, I like what we've got here much better, so good call. Was probably just me being lazy.

File: src/css/custom.css
Comment (LZoog) [reply src/css/custom.css:106]: I could add a sentence in the doc in this PR about CSS changes, I'll do that with my next push.

File: docs/contributing.md
Comment (LZoog) [reply docs/contributing.md]: I used the "note" one and it doesn't look too bad! The "info" one didn't seem quite right and the green one, eh. You can let me know what you think if you don't like it. 😛

image

File: docs/contributing.md
Comment (LZoog) [reply docs/contributing.md:179]: Fixed ;)

File: sidebars.js
Comment (LZoog) [reply sidebars.js:98]: I personally like it at the top-level but at the bottom of the list so it's quite obvious, but if others prefer it to be in a category with "Welcome" I'm fine with that. @k88hudson do you have a preference?

**Comment by tdsmith** [comment]
> Also, should users request the project-nimbus team under "Reviewers" (in which case, how would a user request to be added to this team)? Or, do we expect that users that want to receive notifications have configured them for this repo?

Over at mozilla/data-docs the convention is to explicitly tag a reviewer. I'd worry that tagging the team could be noisy.

Requesting a review requires write access, so if they're not on project-nimbus they should do that first. :) "Ask in #nimbus-project" seems good, though.

**Comment by pdehaan** [comment]
Re:reviewers, what about using a CODEOWNERS file to auto assign reviewers based on globs? 
https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

For example, Tim+Anna are reviewers for any jetstream changes, experimenter core can be reviewed by whomever adds themselves to the codeowners file for whatever paths they volunteer to review/own. I can review PRs that add .svg files.

if we add e2e testing docs, B4hand and I can be notified on changes in those paths.

File: docs/contributing.md
Comment (jodyheavener) [inline docs/contributing.md:208]: 😂

File: docs/contributing.md
Comment (jodyheavener) [inline docs/contributing.md:25]: Maybe this could be more descriptive? Like "From the repository's 'Notifications settings' menu, select 'All activity' to be notified of all changes to this repository"

**Review by jodyheavener — APPROVED** [review]
This is really great. Content is great. I'm biased but it looks extra great in dark mode. I love the image arrangement.

**Comment by jodyheavener** [comment]
Oh while we're here maybe you wanna update or just remove these blurbs about contributing in the main readme? https://github.com/mozilla/experimenter-docs/#using-github

**Comment by LZoog** [comment]
@pdehaan I like that idea. Let's create a follow up issue for that since it might warrant some discussion?

I like Tim's suggestion of tagging someone (or the team) and/or "Ask in #nimbus-project" just for now so I can get this merged in. 😎

**Comment by LZoog** [comment]
OK, made a few changes including:
* I went ahead and made the "Contributing" link fall under a "Welcome" category as suggested by @jodyheavener, I agree this looks nicer
* I added an "info" blurb around sidebar labels
* I added a "Pull Request Workflow" section
* I edited the front-page documentation
* I added an "Experimenter Engineers" section for Jared to start adding docs to (we can re-order stuff if we want, maybe Engineering should all be grouped together somehow)
* I addressed other comments

Gonna go ahead and pull the trigger here. Anything

**Comment by pdehaan** [comment]
> Let's create a follow up issue for that since it might warrant some discussion?

Filed as https://github.com/mozilla/experimenter-docs/issues/11

**Pull request merged**
